core/device.h

```diff
@@ -40,6 +41,12 @@ const char *device_get_nickname(const struct device *dev);
 extern struct device_table *alloc_device_table();
 extern void free_device_table(struct device_table *devices);
 
+// create fingerprint entry - raw data remains owned by caller
+extern void create_fingerprint_node(struct fingerprint_table *table, uint32_t model, uint32_t serial,
+				   const unsigned char *raw_data, unsigned int fsize, uint32_t fdeviceid, uint32_t fdiveid);
+// look up the fingerprint for model/serial - returns the number of bytes in the fingerprint; memory owned by the table
```

Comment:
If the table owns the memory, the last argument probably should be `const unsigned char **`, no?

Reply:
> If the table owns the memory, the last argument probably should be `const unsigned char **`, no?

I'm sure this is a silly question, but... why?
The caller owns the memory, so the caller passes in a pointer to it, and the `vector` creates its own copy.
What am I missing

Reply:
never mind - understood and addressed